pkg/container/linux_container_environment_extensions.go

```diff
@@ -0,0 +1,71 @@
+package container
+
+import (
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type LinuxContainerEnvironmentExtensions struct {
+}
```

Comment:
Shouldn't this implement the `ExecutionsEnvironment`?

Also, why is it called `LinuxContainerEnvironmentExtensions` instead of `LinuxContainerEnvironment`?

Reply:
Okay got it. I looked for a way to have the go compiler inform that the interface was not implemented.
If it is defined as an 'implicit' interface, it does work, but whenever the interface changes it does break and it's hard to understand why. Have seen that several times already.
But moving everything around and making test harder is also not a good option.

Do you see any other idea to make the compiler more aware of the requirements?

Reply:
> Other than assigning a struct to an interface, I don't see how the compiler will alert you about an error by not meeting the interface requirements.

You are right, interfaces are implicit (I'm still learning go...)

This might be nitpicking and I will resolve this, but we could do this either near to the struct or in a test:
```go
type Interf interface {
	Run()
}

type Impl struct{}

var _ Interf = Impl{}
var _ Interf = (*Impl)(nil)
```

Anyway, I'll leave this decision up to you if you see value in it. :smiley: